src/lib.rs

```diff
@@ -14,19 +14,193 @@ use rustty::CellAccessor;
+        true, true, true, false, false, false, false, false];
+    // 1802 = 10 + 7*(2^8), so writing in little endian byte order but writing
+    // the bits within each byte with MSB on the left, we have
+    // [00001010][00000111].
+    // As a BitVec, we want 01010000 11100000.
+
+    #[test]
+    fn test_bitvec_bit_order() {
+        // Consider a two-byte number where the firt byte's value is 10 and
+        // the second byte's value is 7. Converting to a little endian byte
+        // array should make the 0th byte 10 and the 1th byte 7.
+        let ten_seven = BigInt::from(10 + 7*(2.pow(8)));
+        let bytes = ten_seven.to_bytes_le().1;
+        assert_eq!(bytes[0], 10);
+        assert_eq!(bytes[1], 7);
+        // Now check what happens when we convert this to a BitVec. The bytes
+        // are in little endian order, but the bits within each byte are big
+        // endian:
+        // [00001010][00000111]
+        let bits = BitVec::from_bytes(&bytes);
+        let expected = vec![
+            // 0th byte representing 10
+            false,
+            false,
+            false,
+            false,
+            true,
+            false,
+            true,
+            false,
+            // 1st byte representing 7
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            true,
+            true];
+        for i in 0..16 {
+            assert_eq!(bits[i], expected[i]);
+        }
+    }
+
+    #[test]
+    fn test_conway_kode() {
+        let expected = "476348294852520375132009738840824718882889556\
+                        423255282629108876378472743729817205343700177\
+                        683429960362194923168607044012736510546282236\
+                        08960".parse::<BigInt>().unwrap();
+        assert_eq!(expected, conway_kode());
+    }
+
+    #[test]
+    fn test_grid_from_gidney_string() {
+
+        let grid = grid_from_gidney_string(vec!["   ", "   "]);
+        let mut expected = CellSet::new();
+        assert_eq!(grid, expected);
+
+        let grid = grid_from_gidney_string(vec!["#  ", "   "]);
+        expected.insert((0, 0));
+        assert_eq!(grid, expected);
+
+        let grid = grid_from_gidney_string(vec!["#  ", " # "]);
+        expected.insert((1, 1));
+        assert_eq!(grid, expected);
+    }
+
+    #[test]
+    fn test_decimal_encoded_string_to_bitvec() {
+        let s = "1802";
+        let bitvec = decimal_encoded_string_to_bitvec(s);
+        for i in 0..16 {
+            assert_eq!(bitvec[i], EXPECTED_1082_BITS[i]);
+        }
+    }
+
+    #[test]
+    fn test_bigint_to_bitvec(){
+        let v = bigint_to_bitvec(BigInt::from(1802));
+        for i in 0..16 {
+            assert_eq!(v[i], EXPECTED_1082_BITS[i]);
+        }
+    }
+
+    #[test]
+    fn test_get_state() {
+        let mut grid = CellSet::new();
+        //  0
+        // 0#< look here
```

Comment:
I'm not sure what this comment means

Reply:
Yeah this is difficult to represent in ASCII art.